lib/handlers/sender_handler.rb

```diff
@@ -144,6 +149,7 @@ def on_sendable(sender)
         exit_timer.reset if exit_timer
         # Create new message
         msg = Qpid::Proton::Message.new
+        msg.address = @broker.amqp_address if @anonymous
```

Comment:
This can be done in all cases, not if anonymous only, we can discuss.

Reply:
From previous experience, you could break Artemis if you did not set it. (Hawtio browsing bug.) Now it should not matter either way, if we set i or not... Don't really have preference. Do you want me to change this?

Reply:
https://www.youtube.com/watch?v=2xDzVZcqtYI